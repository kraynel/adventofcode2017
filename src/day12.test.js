const { countConnected, countGroups } = require("./day12");
const fs = require("fs");
const _ = require("lodash");

describe("countConnected", () => {
  it("should build a tree", () => {
    expect(
      countConnected(`0 <-> 2
1 <-> 1
2 <-> 0, 3, 4
3 <-> 2, 4
4 <-> 2, 3, 6
5 <-> 6
6 <-> 4, 5`)
    ).toBe(6);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day12.txt", "utf8", (err, data) => {
      expect(countConnected(data)).toMatchSnapshot();
      done();
    });
  });
});

describe("countGroups", () => {
  it("should count the connected groups", () => {
    expect(
      countGroups(`0 <-> 2
1 <-> 1
2 <-> 0, 3, 4
3 <-> 2, 4
4 <-> 2, 3, 6
5 <-> 6
6 <-> 4, 5`)
    ).toBe(2);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day12.txt", "utf8", (err, data) => {
      expect(countGroups(data)).toMatchSnapshot();
      done();
    });
  });
});
