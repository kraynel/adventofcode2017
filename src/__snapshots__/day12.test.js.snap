// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countConnected should get the value 1`] = `145`;

exports[`countGroups should get the value 1`] = `207`;
